libs/nest/problem/src/lib/base-problem.filter.ts

```diff
@@ -37,6 +47,12 @@ export abstract class BaseProblemFilter implements ExceptionFilter {
     const response = ctx.getResponse<Response>()
     const problem = this.getProblem(error)
 
+    if (problem.status && problem.status < 500) {
+      this.options.logAllErrors && this.logger.info(error)
+    } else {
+      this.logger.error(error)
+    }
```

Comment:
```suggestion
    if (problem.status && problem.status >= 500) {
      this.logger.error(error)
    } else if (this.options.logAllErrors) {
       this.logger.info(error)
    }
```

Not a fan of `condition && doStuff` for clarity's sake. Also think it makes sense to reverse the condition to not log all problems with `!problem.status` as errors.

Same above.

Reply:
Good points - will change.